hypothesis-python/src/hypothesis/core.py

```diff
@@ -354,17 +348,18 @@ def report_buffered():
                 # a saved failure when other arguments are supplied by e.g. pytest.
                 # See https://github.com/HypothesisWorks/hypothesis/issues/2125
                 pass
-            except BaseException:
-                report_buffered()
-                raise
+            except BaseException as err:
+                # In order to support reporting of multiple failing examples, we yield
+                # each of the (report text, error) pairs we find back to the top-level
+                # runner.  This also ensures that user-facing stack traces have as few
+                # frames of Hypothesis internals as possible.
+                yield (fragments_reported, err.with_traceback(get_trimmed_traceback()))
+                continue
```

Comment:
Does this not result in us running multiple explicit examples even if we're only reporting one? Should we break instead of continuing here if `report_multiple_failures` is false?

Reply:
Done.